README.md

```diff
@@ -208,3 +208,317 @@ Explore victoriaMetrics go client <https://github.com/VictoriaMetrics/metrics>
 
 
 Handle routing errors properly instead of returning HTTP500 in all cases, eg: sql transaction request must return HTTP400 instead of HTTP500
+
+
+# RZP-Trino REST Client
+
+## Overview
+
+This module exposes an API to interact with Trino. It retrieves data using the Trino Go client, processes it, and returns it in JSON format to the client.
+
+### Features:
+- **Trino Integration:** Connects to a Trino instance and executes SQL queries.
+- **Data Processing:** Applies business logic to the data before sending it to the client.
+- **JSON API:** Communicates with clients using JSON format.
+- **Request/Response Limitation:** Throws an error if response data exceeds 5,000 records.
+- **Configurable:** Uses TOML files for different environments (development, staging, production).
+
+## Base URL
+
+The base URL for the API is: `http://rzp-trino-client:8000`
+
+### 1. `GET /v1/health`
+
+**Description:**  
+Checks the health of the trino client API service.
+
+**Response:**
+
+```json
+{
+  "status": "string",  // Service health status
+}
+```
+
+### 2. `POST /v1/query`
+
+**Description:**  
+Executes a SQL query against Trino.
+
+**Request:**
+
+```json
+{
+  "sql": "string"  // SQL query to be executed on trino
+}
+```
+
+**Reponse:**
+
+```json
+{
+  "status": "string",  // "Success/ Error/ Running"
+  "columns": [
+    {
+      "name": "string",  // Column name
+      "type": "string"   // Column type
+    }
+  ],
+  "data": [
+    [
+      {
+        "data": "string"  // Data for each cell
+      }
+    ]
+  ],
+  "error": {
+    "message": "string",   // Error message (if any)
+    "errorCode": "integer",// Error code (if any)
+    "errorName": "string", // Error name (if any)
+    "errorType": "string"  // Error type (if any)
+  }
+} 
+```
+
+**Reponse(Success example):**
+
+```json
+{
+    "status": "Success",
+    "columns": [
+        {
+            "name": "nationkey",
+            "type": "BIGINT"
+        },
+        {
+            "name": "name",
+            "type": "VARCHAR"
+        },
+        {
+            "name": "regionkey",
+            "type": "BIGINT"
+        },
+        {
+            "name": "comment",
+            "type": "VARCHAR"
+        }
+    ],
+    "data": [
+        [
```

Comment:
json lists are unordered, this wont work

Reply:
we need to use a map/ dict

for instance

```
{
    "column_metadata": [
        {
            "name": "col1",
            "type": "varchar"
        }
    ],
    "data":[
        {
            "col1": "val1"
        },
        {
            "col1": "val2"
        }
    ]
}

```

Reply:
didn't got this, please help me in understanding this more clearly

Reply:
done